File: Data Structures & Algorithms/binary-search/submission-0.java
class Solution {
    public int search(int[] nums, int target) {
        int left = nums[0], right = nums[nums.length - 1];
        while(left <= right){
            int mid = left + ((right - left) / 2);
            if(nums[mid] > target){
                right = mid - 1;
            }else if(nums[mid] < target){
                left = mid + 1;
            }else return mid;
        }
        return -1;
    }
}

File: Data Structures & Algorithms/squares-of-a-sorted-array/submission-0.cs
public class Solution {
    public int[] SortedSquares(int[] nums) {
        int n = nums.Length;
        int[] res = new int[n];
        int left = 0, right = n - 1, resIndex = n - 1;
        while(left <= right){
            if(Math.Abs(nums[left]) < Math.Abs(nums[right])){
                res[resIndex] = nums[right] * nums[right];
                right--;
            }else{
                res[resIndex] = nums[left] * nums[left];
                left++;
            }
            resIndex--;
        }
        return res;
    }
}